### ISSUE TITLE
[🚀 Feature]: Decreasing container layers

### ISSUE THREAD

**Issue by efeozsoy** [SeleniumHQ/docker-selenium#2210]
### Feature and motivation

In our current structure we had to use docker standalone image and latest version has 126 layers which has blocking us to add new things inside the container and taking a docker and push to aws ecr. It gives "rpc error: code = Unknown desc = failed to register layer: max depth exceeded what does it mean"

### Usage example

we are currently planning to add other packages and even automation project inside this image but it blocks to proceed without changing the original project.

**Comment by diemol** [comment]
Several `ARG` and `ENV` lines could probably be grouped in single lines.

**Comment by VietND96** [comment]
If without private info, can you share a sample that you are doing for testing purposes during development?

**Comment by efeozsoy** [comment]
we are using codebuild to push to ecr but for example you can make a simple automation project to inject it, also we are downloading python to run automation.

```
# Use the specified version of the selenium chrome standalone image
FROM selenium/standalone-chrome:123.0

# Set the user to root to install packages
USER root

# Install dependencies necessary for compiling Python
RUN apt-get update && apt-get install -y \
    wget \
    build-essential \
    libssl-dev \
    zlib1g-dev \
    libncurses5-dev \
    libncursesw5-dev \
    libreadline-dev \
    libsqlite3-dev \
    libgdbm-dev \
    libdb5.3-dev \
    libbz2-dev \
    libexpat1-dev \
    liblzma-dev \
    libffi-dev \
    uuid-dev \
    && rm -rf /var/lib/apt/lists/*

# Download and install Python
ENV PYTHON_VERSION=3.10.6

RUN wget https://www.python.org/ftp/python/$PYTHON_VERSION/Python-$PYTHON_VERSION.tgz \
    && tar -xzf Python-$PYTHON_VERSION.tgz \
    && cd Python-$PYTHON_VERSION \
    && ./configure --enable-optimizations \
    && make -j 8 \
    && make altinstall \
    && cd .. \
    && rm -rf Python-$PYTHON_VERSION.tgz Python-$PYTHON_VERSION

# Link the python3 binary to python for convenience
RUN ln -s /usr/local/bin/python3.10 /usr/local/bin/python && \
    ln -s /usr/local/bin/pip3.10 /usr/local/bin/pip

# Switch back to the default "seluser" from selenium base image
USER seluser
COPY . /automation/

CMD ["/opt/bin/entry_point.sh"]
```
it is smth like that i created via chatgpt since i can not provide original codes

**Comment by VietND96** [comment]
Ok, that's fine, I also found a command to count the number of layers
```
$ docker history -q selenium/standalone-chrome:nightly | wc -l
127
```
Just update something and check this number can be decreased

**Comment by VietND96** [comment]
Below is the image layers count after changing. It will be released in 4.20.0

```bash
docker history selenium/base:trunk-20240423 -q | wc -l
29
docker history selenium/hub:trunk-20240423 -q | wc -l
41
docker history selenium/distributor:trunk-20240423 -q | wc -l
38
docker history selenium/router:trunk-20240423 -q | wc -l
37
docker history selenium/sessions:trunk-20240423 -q | wc -l
35
docker history selenium/session-queue:trunk-20240423 -q | wc -l
37
docker history selenium/event-bus:trunk-20240423 -q | wc -l
37
docker history selenium/node-base:trunk-20240423 -q | wc -l
45
docker history selenium/node-chrome:trunk-20240423 -q | wc -l
61
docker history selenium/node-edge:trunk-20240423 -q | wc -l
61
docker history selenium/node-firefox:trunk-20240423 -q | wc -l
58
docker history selenium/node-docker:trunk-20240423 -q | wc -l
37
docker history selenium/standalone-chrome:trunk-20240423 -q | wc -l
72
docker history selenium/standalone-edge:trunk-20240423 -q | wc -l
72
docker history selenium/standalone-firefox:trunk-20240423 -q | wc -l
69
docker history selenium/standalone-docker:trunk-20240423 -q | wc -l
45
docker history selenium/video:ffmpeg-7.0-20240423 -q | wc -l
53
```
